src/main/java/org/folio/linked/data/integration/ResourceModificationEventListener.java

```diff
@@ -51,4 +61,20 @@ public void afterIndex(ResourceIndexedEvent resourceIndexedEvent) {
     resourceRepository.updateIndexDate(resourceIndexedEvent.workId());
   }
 
+  private void sendToSearch(Resource resource) {
+    if (isOfType(resource, WORK)) {
+      kafkaSearchSender.sendSingleResourceCreated(resource);
+    } else {
+      extractWork(resource)
+        .map(Resource::getId)
```

Comment:
Hi @PBobylev,

Do we need  these 2 lines here?

```
        .map(Resource::getId)
        .map(resourceRepository::getReferenceById)
```

We already have `Optional<Work>` in line 68.  (Which is actually retrieved in line 40). No point in fetching the same from DB again right?

Reply:
right, changed